lib/file/getFileBlobAsync.js

```diff
@@ -2,10 +2,24 @@ import { extname } from 'path';
 
 import { ENCODING, FILE_EXTENSION } from './constants';
 
+let wrapBlob = blob => {
+  // Cypress version 5 assigns a function with a compatibility warning
+  // to blob.then, but that makes the Blob actually thenable. We have
+  // to remove that to Promise.resolve not treat it as thenable.
+  //
+  // eslint-disable-next-line no-param-reassign
+  delete blob.then;
+  return Cypress.Promise.resolve(blob);
+};
+
+if (Cypress.version < '5.0.0') {
```

Comment:
This is an interesting way to check versions. However, this might fail once Cypress goes past version 9.

```js
'9.0.0' < '5.0.0' // => false
'10.0.0' < '5.0.0' // => true
```

Is there another way to check this?

Reply:
[The Cypress documentation](https://docs.cypress.io/api/cypress-api/version.html) suggests to use the [semver](https://github.com/npm/node-semver) library.

Reply:
Nice catch!

How would you feel about including this module to do the checking? https://www.npmjs.com/package/compare-versions

Ideally I think it would be good to remove this check once Cypress has updated to remove the change that caused the issue, which hopefully comes in the 5.x.x line and well before 10.x.x.... :crossed_fingers:

Reply:
Thanks @josefsmrz - I'll use that instead :+1:

Reply:
@emilong As I'm thinking about it, I came to another way that might be safer than hinging on Cypress version. What if we check the return type? For example:

```js
let wrapBlob = blob => {
  // Cypress version 5 assigns a function with a compatibility warning
  // to blob.then, but that makes the Blob actually thenable. We have
  // to remove that to Promise.resolve not treat it as thenable.
  //
  // eslint-disable-next-line no-param-reassign
  delete blob.then;
  return blob instanceof Promise ? Cypress.Promise.resolve(blob) : blob;
};
```

This then has no dependencies. Would you be willing to try this?

Reply:
Sure! I probably won't be able to get back to validating that this works until tonight/tomorrow though, but I like that way better!

Reply:
The `version` is available from the plugins `config` since 4.12.1 also https://on.cypress.io/writing-a-plugin#config